### ISSUE TITLE
Fully-Fused MLP implementation

### ISSUE THREAD

**Issue by salykovaa** [intel/intel-extension-for-pytorch#320]
Dear all,

Developers from Nvidia have recently re-implemented MLP network and called it "Fully-Fused MLP", which is 5x-10x faster than the equivalent MLP implementation from PyTorch or TensorFlow. People currently use these fully-fused MLPs  to accelerate training and inference of ML applications, particularly they use them extensively in neural computer graphics to obtain high gains in speed. I would like to know do you maybe plan to implement it or do you have an equivalent implementation? 

I think it is a nice feature, which will enrich the intel extension and make it more attractive for machine learning / computer graphics researchers and engineers.

<img width="847" alt="image" src="https://user-images.githubusercontent.com/63703454/229091825-8126ac7f-7c1e-4e46-b7dc-599696b5d030.png">

References:
[1] [Real-time Neural Radiance Caching for Path Tracing](https://arxiv.org/abs/2106.12372)
[2] [Technical Paper](https://www.nvidia.com/en-us/on-demand/session/gtcspring21-e31307/)
[3] [Nvidia's Fully fused MLP implementation](https://github.com/NVlabs/tiny-cuda-nn)

**Comment by salykovaa** [comment]
Hi @jingxu10 

Should it not be labeled as XPU/GPU and not as CPU, since it is a GPU related implementation?

**Comment by jingxu10** [comment]
Thanks for the clarification.
This implementation seems to be device agnostic, label it with both CPU and GPU.

**Comment by gujinghui** [comment]
@salykovaa  Thanks for the information. We will take it into account.

**Comment by gujinghui** [comment]
@arthuryuan1987

**Comment by sanchitintel** [comment]
@gujinghui, oneDNN Graph Compiler backend supports fusion pattern of multiple MLP layers. I'm not sure if they currently support GPU backend as well.

**Comment by Yunaik** [comment]
@salykovaa we're excited to announce the release of tiny-dpcpp-nn that provides fully-fused MLPs and the same interfaces as tiny-cuda-nn: [tiny-dpcpp-nn](https://github.com/intel/tiny-dpcpp-nn/)

**Comment by salykovaa** [comment]
@Yunaik Thank you for your reply! I will check it

**Comment by jgong5** [comment]
> @salykovaa we're excited to announce the release of tiny-dpcpp-nn that provides fully-fused MLPs and the same interfaces as tiny-cuda-nn: [tiny-dpcpp-nn](https://github.com/intel/tiny-dpcpp-nn/)

Nice work. Just wondering: Do you have plan to contribute the kernels to IPEX?

**Comment by cbauinge** [comment]
> > @salykovaa we're excited to announce the release of tiny-dpcpp-nn that provides fully-fused MLPs and the same interfaces as tiny-cuda-nn: [tiny-dpcpp-nn](https://github.com/intel/tiny-dpcpp-nn/)
> 
> Nice work. Just wondering: Do you have plan/interest to contribute the kernels to IPEX?

@jgong5 There are no plans yet to include it in IPEX but I would be delighted to implement our approach. Ping me if you want to discuss it in more detail.
